src/app/Fake.DotNet.Fsc/Fsc.fs

```diff
@@ -0,0 +1,449 @@
+    let optsArr = makeArgsList opts srcFiles
+
+    Trace.trace <| sprintf "FSC with args:%A" optsArr
+    let errors, exitCode = compiler optsArr
+    
+    reportErrors errors
+    exitCode
+
+/// Common compiler arg prep code
+let private doCompile (compiler: CompilerFunc) (fscParams : FscParam list) (inputFiles : string list) : int = 
+    let inputFiles = inputFiles |> Seq.toList
+    let taskDesc = inputFiles |> FakeString.separated ", "
+    let fscParams = if fscParams = [] then FscParam.Defaults else fscParams
+    let argList = fscParams |> List.map string
+
+    use __ = Trace.traceTask "Fsc " taskDesc
+    let res = compileFiles compiler inputFiles argList
+    res
+
+
+/// The internal FCS Compiler
+let private scsCompile optsArr = 
+    let scs = FSharpChecker.Create()
+    // Always prepend "fsc.exe" since fsc compiler skips the first argument
+    let optsArr = Array.append [|"fsc.exe"|] optsArr
+    let errors, exitcode = scs.Compile optsArr |> Async.RunSynchronously
+
+    /// Better compile reporting thanks to:
+    /// https://github.com/jbtule/ComposableExtensions/blob/5b961b30668bb7f4d17238770869b5a884bc591f/tools/CompilerHelper.fsx#L233
+    let errors = errors |> Array.map (fun (e: FSharpErrorInfo) -> 
+        match e.Severity with
+            | FSharpErrorSeverity.Error -> FscResultMessage.Error e.Message
+            | FSharpErrorSeverity.Warning -> FscResultMessage.Warning e.Message)
+
+    errors, exitcode
+
+/// Compiles the given F# source files with the specified parameters.
+///
+/// ## Parameters
+///
+///  - `setParams` - Function used to overwrite the default Fsc parameters.
+///  - `inputFiles` - The F# input files.
+///
+/// ## Returns
+///
+/// The exit status code of the compile process.
+///
+/// ## Sample
+///
+///     ["file1.fs"; "file2.fs"]
+///     |> compile [Out "" 
+///                 Target Exe
+///                 Platform AnyCpu
+///                 References []
+///                 Debug false 
+///             ]
+let compile (fscParams : FscParam list) (inputFiles : string list) : int = 
+    doCompile scsCompile fscParams inputFiles
+
+/// Compiles one or more F# source files with the specified parameters.
+/// ## Parameters
+///
+///  - `setParams` - Function used to overwrite the default Fsc parameters.
+///  - `inputFiles` - The F# input files.
+///
+/// ## Sample
+///
+///     ["file1.fs"; "file2.fs"]
+///     |> Compile [Out "" 
+///                 Target Exe
+///                 Platform AnyCpu
+///                 References []
+///                 Debug false 
+///             ]
+let Compile (fscParams : FscParam list) (inputFiles : string list) : unit = 
+    let res = compile fscParams inputFiles
+    if res <> 0 then raise <| BuildException("Fsc: compile failed with exit code", [ string res ])
+
+
+/// An externally referenced compiler
+let private externalCompile (fscTool: string) (optsArr: string []) = 
+    let args = String.Join(" ", optsArr)
```

Comment:
I think we should use `(Arguments.OfArgs optsArr).ToWindowsCommandLine` instead of `String.Join` the former will handle arguments with spaces.

Reply:
Hey I'm getting a weird name conflict error when trying to use Arguments. In vscode, everything is fine, but when running build.fsx, and in visual studio, the compiler thinks I'm trying to use the discriminated union case `DocoptResult.Arguments` instead of the one defined in `Fake.Core.Process`.
  
The error im getting specifically is
>  error FS0072: Lookup on object of indeterminate type based on information prior to this program point. A type annotation may be needed prior to this program point to constrain the type of the object. This may
allow the lookup to be resolved.

Both of these Argument types exist in Fake.Core so I'm not really sure how I can inform the compiler of which one it needs to use. Any ideas on how to fix this?

Reply:
Nevermind, figured it out. The file with Arguments defined in Fake.Core.Process was not included in the fakelib.fsproj config

Reply:
I just fixed the same in release/rc_12 branch